Professional-agentic-architect-section4.md

```
@@ -0,0 +1,497 @@
+    loop 約10分ごと
+        Agent->>Trace: OpenTelemetry形式の\nトレース・ログを出力
+        Monitor->>Trace: サンプリング条件に基づき\nトレースをクエリ
+        Monitor->>Eval: サンプリングしたトレースを\n設定済みメトリクスで評価
+        Eval-->>Monitor: スコア・根拠を返却
+        Monitor->>Out: 結果を書き込み・\n数値スコアをエクスポート
+    end
+```
+
+Online Monitorを機能させるには、エージェント側が特定のOpenTelemetryシグナルをCloud Traceに出力している必要があります。具体的には、エージェント名・説明・会話IDを含む「invoke agentスパン」と、プロンプト・応答・システム指示・ツール定義を含む `gen_ai.client.inference.operation.details` イベントです[^6]。ADKを使っている場合は、次の環境変数を設定することでこれらのテレメトリを有効化します[^6]。
+
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)